controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -79,291 +75,318 @@ func (r *KafkaUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		l.Error(err, "Unable to fetch Kafka User", "request", req)
 		return models.ReconcileRequeue, nil
 	}
+	secret := &v1.Secret{}
+	kafkaUserSecretNamespacedName := types.NamespacedName{
+		Name:      user.Spec.SecretRef.Name,
+		Namespace: user.Spec.SecretRef.Namespace,
+	}
+	err = r.Client.Get(ctx, kafkaUserSecretNamespacedName, secret)
+	if k8serrors.IsNotFound(err) {
+		l.Error(err, "Cannot get Kafka User credentials. Secret is not found",
+			"request", req,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.FetchFailed,
+			"Fetch user credentials secret is failed. Reason: %v",
+			err,
+		)
 
-	switch user.Annotations[models.ResourceStateAnnotation] {
-	case models.CreatingEvent:
-		return r.handleCreateKafkaUser(ctx, user, l), nil
+		return models.ReconcileRequeue, nil
+	}
+	username, password, err := r.getKafkaUserCredsFromSecret(user.Spec)
+	if err != nil {
+		l.Error(
+			err, "Cannot get Kafka User creds from secret",
+			"kafka user spec", user.Spec,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.FetchFailed,
+			"Fetch user credentials from secret is failed. Reason: %v",
+			err,
+		)
 
-	case models.UpdatingEvent:
-		return r.handleUpdateKafkaUser(ctx, user, l), nil
+		return models.ReconcileRequeue, nil
+	}
+	patch := user.NewPatch()
+	if controllerutil.AddFinalizer(user, user.GetDeletionFinalizer()) {
+		err = r.Patch(ctx, user, patch)
+		if err != nil {
+			l.Error(err, "Cannot patch Kafka user with deletion finalizer")
+			r.EventRecorder.Eventf(
+				user, models.Warning, models.PatchFailed,
+				"Patching Kafka user with deletion finalizer has been failed. Reason: %v", err,
+			)
 
-	case models.DeletingEvent:
-		return r.handleDeleteKafkaUser(ctx, user, l), nil
-	case models.SecretEvent:
-		kafkaUserSecret := &v1.Secret{}
-		kafkaUserSecretNamespacedName := types.NamespacedName{
-			Name:      user.Spec.KafkaUserSecretName,
-			Namespace: user.Spec.KafkaUserSecretNamespace,
+			return models.ReconcileRequeue, nil
 		}
-		err = r.Client.Get(ctx, kafkaUserSecretNamespacedName, kafkaUserSecret)
-		if k8serrors.IsNotFound(err) {
-			l.Error(err, "Cannot get Kafka User credentials. Secret is not found",
-				"request", req,
+	}
+
+	for clusterID, clusterEvent := range user.Status.ClustersEvents {
+		if clusterEvent == models.CreatingEvent {
+			l.Info(
+				"Creating Kafka User resource",
```

Comment:
Creating Kafka User resource -> Creating Kafka User

Fix in all similar cases please

Reply:
done